### PR TITLE
Python 2/3 (and some spelling corrections).

### PR DESCRIPTION
**Pull request by DailyDreaming**
Building off of some of the comments in #2338.

### CODE REVIEW

**Comment by mr-c** [comment]
Does this turn on Py3 testing on Jenkins/TravisCI?

**Comment by DailyDreaming** [comment]
Pushed a commit that isn't showing?
http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/2916/

**Comment by DailyDreaming** [comment]
Pushed a commit that isn't showing?
http://jenkins.cgcloud.info/job/toil-pull-request-integration-tests/2916/
http://jenkins.cgcloud.info/job/toil-pull-requests/4621/

**Comment by DailyDreaming** [comment]
Jenkins, test this please.

**Comment by DailyDreaming** [comment]
Jenkins, test this please.

**Comment by DailyDreaming** [comment]
Jenkins, test this please.

**Comment by DailyDreaming** [comment]
```
WARNING:toil.leader:q/Y/job0YFWiV    ERROR:cwltool:Got workflow error
WARNING:toil.leader:q/Y/job0YFWiV    Traceback (most recent call last):
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/venv/local/lib/python2.7/site-packages/cwltool/executors.py", line 144, in run_jobs
WARNING:toil.leader:q/Y/job0YFWiV        for job in jobiter:
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/venv/local/lib/python2.7/site-packages/cwltool/command_line_tool.py", line 405, in job
WARNING:toil.leader:q/Y/job0YFWiV        reffiles, builder.stagedir, runtimeContext, True)
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/src/toil/cwl/cwltoil.py", line 273, in make_path_mapper
WARNING:toil.leader:q/Y/job0YFWiV        runtimeContext.toil_get_file)
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/src/toil/cwl/cwltoil.py", line 227, in __init__
WARNING:toil.leader:q/Y/job0YFWiV        referenced_files, basedir, stagedir, separateDirs=separateDirs)
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/venv/local/lib/python2.7/site-packages/cwltool/pathmapper.py", line 220, in __init__
WARNING:toil.leader:q/Y/job0YFWiV        self.setup(dedup(referenced_files), basedir)
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/venv/local/lib/python2.7/site-packages/cwltool/pathmapper.py", line 273, in setup
WARNING:toil.leader:q/Y/job0YFWiV        self.visit(fob, stagedir, basedir, copy=fob.get("writable"), staged=True)
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/src/toil/cwl/cwltoil.py", line 254, in visit
WARNING:toil.leader:q/Y/job0YFWiV        resolved = self.get_file(loc) if self.get_file else loc
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/src/toil/cwl/cwltoil.py", line 308, in toil_get_file
WARNING:toil.leader:q/Y/job0YFWiV        src_path = file_store.readGlobalFile(file_store_id[7:])
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/src/toil/fileStore.py", line 1661, in readGlobalFile
WARNING:toil.leader:q/Y/job0YFWiV        self.jobStore.readFile(fileStoreID, localFilePath, symlink=symlink)
WARNING:toil.leader:q/Y/job0YFWiV      File "/mnt/ephemeral/workspace/toil-pull-requests/src/toil/jobStores/fileJobStore.py", line 392, in readFile
WARNING:toil.leader:q/Y/job0YFWiV        os.link(jobStoreFilePath, localFilePath)
WARNING:toil.leader:q/Y/job0YFWiV    OSError: [Errno 18] Invalid cross-device link
```
Note: change to `shutil.move()`.

File: src/toil/batchSystems/mesos/executor.py [hunk 1]
```diff
@@ -242,17 +242,20 @@ def main():
...
         log.warning("Had to fake MESOS_AGENT_ENDPOINT as %s" % os.environ["MESOS_AGENT_ENDPOINT"])
-        
-    try:
-        urlopen("http://%s/logging/toggle?level=1&duration=15mins" % os.environ["MESOS_AGENT_ENDPOINT"]).read()
-        log.debug("Toggled agent log level")
-    except Exception as e:
-        log.debug("Failed to toggle agent log level")
+
+    # must be set manually to enable toggling of the mesos log level for debugging jenkins
+    # may be useful: https://github.com/DataBiosphere/toil/pull/2338
```
[the remark this answers is not in the document]
Comment (DailyDreaming) [reply src/toil/batchSystems/mesos/executor.py]: Should be addressed, thanks!

**Comment by adamnovak** [comment]
Any idea why Jenkins is upset about missing `succeeded.log` files? Do those tests just fail sometimes?

**Comment by DailyDreaming** [comment]
@adamnovak Apparently?  Nothing changed in this PR that would cause it to fail, and it obviously passed in the previous commit.  The `succeeded.log`/`failed.log` files have caused a few tests to fail in the past randomly, but we'd thought those were fixed.  I suspect I need to look closer at the status logging to see what's happening.

I feel alright about merging this.  Python 3 tests should pass locally with this PR.  I'll do some python 3 testing with some of our variant calling TOPMed workflows, and make a PR for whatever comes up there.  Then the plan is to move to gitlab.  When that's done I think we can declare it's fully python 3 complaint.

Referencing the failing tests: http://jenkins.cgcloud.info/job/toil-pull-requests/4674/

**Pull request merged**